extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1126,6 +1128,11 @@ export class AmpA4A extends AMP.BaseElement {
           // modified url.
           'src': xhrFor(this.win).getCorsUrl(this.win, adUrl),
         }, SHARED_IFRAME_PROPERTIES));
+    const attributes = this.generateSentinelAndContext(iframe);
+    iframe.setAttribute('name', JSON.stringify(attributes));
```

Comment:
Can you move this logic up so that `'name'` is also set by `createElementWithAttributes`, above?